### PR TITLE
Update chicoma-cpu modules

### PR DESCRIPTION
**Pull request by xylar**
Following the recent DST, this merge updates the module files and environment variables on Chicoma-CPU.  We note that these updates work well for `gnu` and `nvidia` compilers but not yet for `intel`, which we are continuing to work on.  A separate update will be needed to address Chicoma-GPU as well.

### CODE REVIEW

**Comment by xylar** [comment]
This is just a draft so far.  I'm having no luck with either gnu or intel on Chicoma-CPU so far.  I haven't tried anything else yet.

**Comment by xylar** [comment]
I've contacted LANL IC about the trouble I'm having with gnu:
```
/lustre/scratch5/xylar/E3SM/scratch/chicoma-cpu/SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_gnu.20241011_152619_r3zumf/bld/e3sm.exe: /opt/cray/pe/gcc-libs/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /lustre/scratch5/xylar/E3SM/scratch/chicoma-cpu/SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_gnu.20241011_152619_r3zumf/bld/e3sm.exe)
```
While it seems clear that there's an RPATH being set to `/opt/cray/pe/gcc-libs`, I haven't been able to track down where that's coming from.  Setting the `LD_LIBRARY_PATH` didn't help.

**Comment by xylar** [comment]
On the intel side, it's not finding NetCDF-C or -Fortran, even though we're passing a `NETCDF_PATH` environment variable that seems correct.

**Comment by xylar** [comment]
The gnu issue seems similar to https://github.com/E3SM-Project/E3SM/issues/6677

**Comment by jonbob** [comment]
With the commits I just pushed, I was able to successfully build and run:
* SMS.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_gnu
* SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_gnu

So I think at this point we can say we support gnu on chicoma. I'll poke around at intel as well

File: cime_config/machines/config_machines.xml [hunk 1]
```diff
@@ -4215,7 +4215,7 @@ commented out until "*** No rule to make target '.../libadios2pio-nm-lib.a'" iss
     <DIN_LOC_ROOT_CLMFORC>/usr/projects/e3sm/inputdata/atm/datm7</DIN_LOC_ROOT_CLMFORC>
     <DOUT_S_ROOT>/lustre/scratch5/$ENV{USER}/E3SM/archive/$CASE</DOUT_S_ROOT>
     <BASELINE_ROOT>/lustre/scratch5/$ENV{USER}/E3SM/input_data/ccsm_baselines/$COMPILER</BASELINE_ROOT>
-    <CCSM_CPRNC>/usr/projects/climate/SHARED_CLIMATE/software/badger/cprnc</CCSM_CPRNC>
+    <CCSM_CPRNC>/usr/projects/e3sm/software/chicoma-cpu/cprnc</CCSM_CPRNC>
```
Comment (xylar) [inline cime_config/machines/config_machines.xml:4218]: Oh, I bet I know what happened here.  I deleted this thinking that it was old and no longer used.  In my defense, it has `badger` in the path...

File: cime_config/machines/config_machines.xml [hunk 2]
```diff
@@ -4251,26 +4251,27 @@ commented out until "*** No rule to make target '.../libadios2pio-nm-lib.a'" iss
...
         <command name="unload">craype-accel-nvidia80</command>
         <command name="unload">craype-accel-host</command>
         <command name="unload">perftools-base</command>
         <command name="unload">perftools</command>
         <command name="unload">darshan</command>
+        <command name="unload">PrgEnv-gnu</command>
+        <command name="unload">PrgEnv-intel</command>
+        <command name="unload">PrgEnv-nvidia</command>
+        <command name="unload">PrgEnv-cray</command>
+        <command name="unload">PrgEnv-aocc</command>
```
Comment (xylar) [inline cime_config/machines/config_machines.xml:4270]: I found that these needed to be unloaded after their corresponding compiler modules or there would be an error about an undefined environment variable name.

**Comment by jonbob** [comment]
@xylar -- let me take a peek

**Comment by jonbob** [comment]
@xylar - it seems to be struggling with the atm data? That doesn't make much sense

**Comment by xylar** [comment]
In the meantime, I'm trying an optimized run to see how that goes.

**Comment by xylar** [comment]
`SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_gnu` passed for me in the end.  It just took 25 minutes and didn't get to time stepping for a *long* time.  It seems like it might be a file system issue.

**Comment by xylar** [comment]
`SMS.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_gnu` passed for me as well.

**Comment by xylar** [comment]
I tested `SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_nvidia` and it built fine and appeared to be running but timed out before the 30 minutes I gave it (same file system issues as above).

File: cime_config/machines/config_machines.xml (hunk 1 above)
Comment (jonbob) [reply cime_config/machines/config_machines.xml:3940]: And it no longer exists! Not just the machine but that file. But I think if that line is missing it forces each test to try to build it

Comment (xylar) [reply cime_config/machines/config_machines.xml:3940]: > And it no longer exists! 

That's what I was saying.  I think I deleted it trying to free up space in `/usr/projects/climate` because I couldn't imagine we were still using software built for Badger.

**Comment by xylar** [comment]
I realize it's not a high priority for us but `SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_nvidia` passed for me with a longer job runtime.

File: cime_config/machines/config_machines.xml [hunk 3]
```diff
@@ -3973,39 +3973,40 @@ commented out until "*** No rule to make target '.../libadios2pio-nm-lib.a'" iss
...
+        <command name="load">PrgEnv-gnu/8.5.0</command>
         <command name="load">gcc/12.2.0</command>
         <command name="load">cray-libsci/23.05.1.4</command>
       </modules>
 
       <modules compiler="nvidia">
-        <command name="load">PrgEnv-nvidia/8.4.0</command>
-        <command name="load">nvidia/22.7</command>
+        <command name="load">PrgEnv-nvidia/8.5.0</command>
+        <command name="load">nvidia/24.7</command>
```
Comment (xylar) [inline cime_config/machines/config_machines.xml:4003]: I successfully tested these updated modules as well.

File: components/mpas-framework/Makefile [hunk 4]
```diff
@@ -396,11 +396,11 @@ gnu-cray:
 	"FFLAGS_OPT = -O3 -m64 -ffree-line-length-none -fconvert=big-endian -ffree-form -ffpe-summary=none $${EXTRA_FFLAGS}" \
 	"CFLAGS_OPT = -O3 -m64" \
 	"CXXFLAGS_OPT = -O3 -m64" \
-	"LDFLAGS_OPT = -O3 -m64" \
+	"LDFLAGS_OPT = -O3 -m64 $(GNU_CRAY_LDFLAGS)" \
```
Comment (xylar) [inline components/mpas-framework/Makefile:399]: @matthewhoffman, this environment variable (or argument to `make`) needs to be set to:
```
export GNU_CRAY_LDFLAGS="-Wl,--enable-new-dtags"
```
on Chicoma for now.  I'll make sure Compass and Polaris do this.  If someone is building for Chicoma outside of Compass or Polaris (good luck!), they would need to set this manually.

Are you okay with this fix?  I don't want to put in anything into the Makefile that tries to detect the machine or anything crazy like that.

File: cime_config/machines/config_machines.xml [hunk 5]
```diff
@@ -4085,11 +4100,11 @@ AMD EPYC 7713 64-Core (Milan) (256GB) and 4 nvidia A100'</DESC>
...
       <!-- does not exist -->
-      <init_path lang="python">/usr/share/lmod/8.3.1/init/python</init_path>
-      <init_path lang="sh">/usr/share/lmod/8.3.1/init/sh</init_path>
-      <init_path lang="csh">/usr/share/lmod/8.3.1/init/csh</init_path>
+      <init_path lang="python">/usr/share/lmod/lmod/init/python</init_path>
+      <init_path lang="sh">/usr/share/lmod/lmod/init/sh</init_path>
+      <init_path lang="csh">/usr/share/lmod/lmod/init/csh</init_path>
       <cmd_path lang="perl">/usr/share/lmod/lmod/libexec/lmod perl</cmd_path>
       <cmd_path lang="python">/usr/share/lmod/lmod/libexec/lmod python</cmd_path>
       <cmd_path lang="sh">module</cmd_path>
```
Comment (xylar) [inline cime_config/machines/config_machines.xml:4110]: The chicoma-gpu changes should probably be reverted and should be addressed in a separate PR.  I don't want testing these to hold up the chicoma-cpu fixes.

**Comment by xylar** [comment]
@jonbob, at the risk of delaying this further, I think we probably want to follow what Noel is doing on Perlmutter:
https://github.com/E3SM-Project/E3SM/pull/6702/files
That should at least save us from having to make yet another PR in the near future.

File: cime_config/machines/config_machines.xml [hunk 6]
```diff
@@ -3973,39 +3973,42 @@ commented out until "*** No rule to make target '.../libadios2pio-nm-lib.a'" iss
         <command name="unload">cray-parallel-netcdf</command>
         <command name="unload">cray-netcdf</command>
         <command name="unload">cray-hdf5</command>
```
Comment (xylar) [inline cime_config/machines/config_machines.xml:3975]: I tried removing `cpe` like Noel did here:
https://github.com/E3SM-Project/E3SM/blob/bdcc2f551cfae2fca53bd8aa4ec604601ddf1c68/cime_config/machines/config_machines.xml#L193
But I got nasty error like:
```
        Lmod has detected the following error: These module(s) or extension(s) exist
        but cannot be loaded as requested: "git", "cmake/3.27.7"
           Try: "module spider git cmake/3.27.7" to see how to load the module(s).

```

**Comment by xylar** [comment]
I have `gnu` and `nvidia` tests in the queue with the latest updates.

File: components/mpas-framework/Makefile (hunk 4 above)
Comment (matthewhoffman) [reply components/mpas-framework/Makefile:399]: @xylar , this seems like the best solution given the circumstances

**Comment by xylar** [comment]
Closed in favor of https://github.com/E3SM-Project/E3SM/pull/6705
